lib/fog/libvirt/models/compute/server.rb

```diff
@@ -215,6 +219,35 @@ def vnc_port
           display[:port]
         end
 
+        def generate_config_iso(user_data, &blk)
+          Dir.mktmpdir('config') do |wd|
+            generate_config_iso_in_dir(wd, user_data, &blk)
+          end
+        end
+
+        def generate_config_iso_in_dir(dir_path, user_data, &blk)
+          FileUtils.touch(File.join(dir_path, "meta-data"))
+          File.open(File.join(dir_path, 'user-data'), 'w') { |f| f.write user_data }
+
+          isofile = Tempfile.new(['init', '.iso']).path
+          unless system("genisoimage -output #{isofile} -volid cidata -joliet -rock #{File.join(dir_path, 'user-data')} #{File.join(dir_path, 'meta-data')}")
+            raise Fog::Errors::Error("Couldn't generate cloud-init iso disk.")
+          end
+          blk.call(isofile)
+        end
+
+        def create_user_data_iso
+          generate_config_iso(user_data) do |iso|
+            vol = service.volumes.create(:name => cloud_init_volume_name, :capacity => "#{File.new(iso).size}b", :allocation => "0G")
+            vol.upload_image(iso)
+            @iso_file = cloud_init_volume_name
+          end
+        end
+
+        def cloud_init_volume_name
+          "#{name}-cloud-init.iso"
```

Comment:
As opposed to user supplying the name for the user-data iso? Probably doable via some additional constructor parameter, but why would anyone be interested in something like that? 

No idea re: convention, but I figured this is pretty descriptive a name.